### PR TITLE
Setup newrelic

### PR DESCRIPTION
**Pull request by rodolfo3**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -3,11 +3,35 @@ package main
...
 
+func mustGetEnv(key string) string {
+	if val := os.Getenv(key); "" != val {
+		return val
+	}
+	panic(fmt.Sprintf("environment variable %s unset", key))
+}
+
 func main() {
+  nrCfg := newrelic.NewConfig(mustGetEnv("NEW_RELIC_APP_NAME"), mustGetEnv("NEW_RELIC_LICENSE_KEY"))
```
Comment (lucasgomide) [inline main.go]: Please use the [gofmt](https://golang.org/cmd/gofmt/) command to format your code.

Comment (lucasgomide) [inline main.go]: Why are you forcing a lib open-source to use *New Relic* ?

**Review by guilhermearaujo — CHANGES_REQUESTED** [review]
Indentation is all wrong. Go has some tools that fix is automagically, depending on what editor you're using.

File: main.go [hunk 2]
```diff
@@ -3,11 +3,35 @@ package main
...
+  "os"
+  "fmt"
 	"log"
 	"net/http"
+
+  newrelic "github.com/newrelic/go-agent"
+)
+
+var (
+	app newrelic.Application
```
Comment (lucasgomide) [inline main.go]: This is a `package level vars` only accessible on main package.

There is a good [post](http://peter.bourgon.org/blog/2017/06/09/theory-of-modern-go.html) about it.

Do you really does it ?

**Review by lucasgomide — CHANGES_REQUESTED** [review]
use gofmt

File: main.go (hunk 1 above)
Comment (rodolfo3) [reply main.go]: @lucasgomide do you have any other idea or how can we monitoring it, then?

Comment (guilhermearaujo) [reply main.go]: @lucasgomide this is our fork, not the original, open-source repo.

File: Godeps/Godeps.json [hunk 3]
```diff
@@ -35,6 +35,11 @@
 		{
 			"ImportPath": "gopkg.in/yaml.v2",
 			"Rev": "a83829b6f1293c91addabc89d0571c246397bbf4"
-		}
+		},
+    {
+      "ImportPath": "github.com/newrelic/go-agent",
+      "Comment": "v1.9.0",
+      "Rev": "29ec3cd1bb2f21d21d36da37dae52695cb2c3a17"
+    }
```
Comment (guilhermearaujo) [inline Godeps/Godeps.json:43]: This is still not indented

File: main.go [hunk 4]
```diff
@@ -1,21 +1,57 @@
...
+	if os.Getenv("NEW_RELIC_LICENSE_KEY") == "" {
+		return nil
+	}
+
+	nrCfg := newrelic.NewConfig(mustGetEnv("NEW_RELIC_APP_NAME"), mustGetEnv("NEW_RELIC_LICENSE_KEY"))
+	nrCfg.Logger = newrelic.NewDebugLogger(os.Stdout)
+
+	var err error
+	var app newrelic.Application
+	app, err = newrelic.NewApplication(nrCfg)
```
Comment (lucasgomide) [inline main.go:31]: You can initialize those variables only with `:=`
then remove this
```golang
var err error
var app newrelic.Application
```

File: main.go [hunk 5]
```diff
@@ -1,21 +1,57 @@
...
 )
 
+func mustGetEnv(key string) string {
+	if val := os.Getenv(key); "" != val {
+		return val
+	}
+	panic(fmt.Sprintf("environment variable %s unset", key))
+}
+
+func getApp() newrelic.Application {
```
Comment (lucasgomide) [inline main.go:21]: you can rename this method to `setupNewRelic` to explicit what it does

File: main.go [hunk 6]
```diff
@@ -1,21 +1,57 @@
...
 	cfg, err := config.NewConfig()
 	if err != nil {
 		panic(err)
 	}
 
 	address := "0.0.0.0:" + cfg.Port
 	log.Printf("Running on %s", address)
-	err = http.ListenAndServe(address, &handler.RedirectHandler{Client: cfg.RedisClient, Logger: cfg.Logger })
+	if app != nil {
+		http.HandleFunc(newrelic.WrapHandleFunc(app, "/", (&handler.RedirectHandler{Client: cfg.RedisClient, Logger: cfg.Logger}).ServeHTTP))
```
Comment (lucasgomide) [inline main.go:50]: This is bad. But we can improve after it.

**Pull request closed without merging**